src/main/java/org/opensrp/web/controller/UniqueIdController.java

```diff
@@ -117,44 +118,36 @@
 		}
 		
 		return new ResponseEntity<>(new Gson().toJson("" + message), HttpStatus.OK);
-	}
-	
-	boolean checkRoleIfRoleExitst(List<String> roleList, String role) {
-		for (String roleName : roleList)
-			if (StringUtils.containsIgnoreCase(roleName, role))
-				return true;	
-		return false;
-	}
-	
+	} 
 	/**
-	 * Fetch unique Ids from OMRS
+	 * Fetch unique Ids
 	 *
 	 * @return json array object with ids
 	 */
-	
 	@RequestMapping(value = "/get", method = RequestMethod.GET, produces = { MediaType.APPLICATION_JSON_VALUE })
-	protected ResponseEntity<String> get(HttpServletRequest request, Authentication authentication)
-	    throws JsonProcessingException, JSONException {
+	protected ResponseEntity<String> get(HttpServletRequest request, Authentication authentication) throws JSONException {
 		
 		String numberToGenerate = getStringFilter("numberToGenerate", request);
 		String source = getStringFilter("source", request);
 		String usedBy = authentication.getName();
 		Map<String, Object> map = new HashMap<>();
 		IdentifierSource identifierSource = identifierSourceService.findByIdentifier(source);
-		
-		if (identifierSource != null) {
-			List<String> identifiers = uniqueIdentifierService.generateIdentifiers(identifierSource,
-			    Integer.parseInt(numberToGenerate), usedBy);
-			map.put("identifiers", identifiers);
-			
+		try {
+			if (identifierSource != null) {
+				List<String> identifiers = uniqueIdentifierService.generateIdentifiers(identifierSource,
+				    Integer.parseInt(numberToGenerate), usedBy);
+				if (identifiers != null) {
+					map.put("identifiers", identifiers);
+				}
+			} else {
+				map.put("identifiers", openmrsIdService.getOpenMRSIdentifiers(source, numberToGenerate));
+			}
 			return new ResponseEntity<>(objectMapper.writeValueAsString(map), HttpStatus.OK);
-			
-		} else {
-			
-			return new ResponseEntity<>(String.format("Unique Identifier source %s not found", source),
-			        HttpStatus.NOT_FOUND);
 		}
-		
+		catch (IllegalArgumentException | JsonProcessingException exception) {
+			return new ResponseEntity<>(exception.getMessage(), HttpStatus.BAD_REQUEST);
+		}
```

Comment:
These are handled by global exception handler

Reply:
Do we get rid of the whole try catch?

Reply:
yes, it was added on this PR

Reply:
Looks like overwrites the earlier PR of removing OpenSRP ID gen, you should retain what is in master

Reply:
gotchu

Reply:
Gotchu